src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -12,9 +12,23 @@ namespace Stratis.Bitcoin.Consensus
     /// <summary>
     /// TODO comment this interface
     /// </summary>
-    public interface IChainedHeaderValidator
+    public interface IChainedHeaderTreeValidator
     {
-        void Validate(ChainedHeader chainedHeader);
+        /// <summary>
+        /// Validation of a header that was seen for the first time.
```

Comment:
`Validation of a header that was seen for the first time.`

So the header that was seen before can't be validated using this method?

Reply:
It could but it wont (according to the code), but the comment does no suggest a validated header cannot be validated again.